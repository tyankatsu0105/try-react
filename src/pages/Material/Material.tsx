import React, { FunctionComponent } from 'react';
import { Helmet } from 'react-helmet';
import Button from '@material-ui/core/Button';

const Material: FunctionComponent = () => {
  return (
    <>
      <Helmet>
        <title>This is Material</title>
      </Helmet>

      <Button variant="contained" color="primary">
        Hello World
      </Button>
    </>
  );
};

export default Material;
