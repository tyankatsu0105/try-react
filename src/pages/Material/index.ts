import { lazy } from 'react';

export const Material = lazy(() => import('./Material'));
