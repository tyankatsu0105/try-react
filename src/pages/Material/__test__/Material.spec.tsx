import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';

import Material from '../Material';

describe('Material', () => {
  it('snapshot', () => {
    const { baseElement } = render(
      <MemoryRouter initialEntries={['/']}>
        <Material />
      </MemoryRouter>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
