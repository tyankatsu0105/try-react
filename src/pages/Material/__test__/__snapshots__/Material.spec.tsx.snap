// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Material snapshot 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Hello World
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;
